src/WordAnalyzer.ts

```diff
@@ -0,0 +1,105 @@
+import * as unzipper from 'unzipper';
+import * as fs from 'fs';
+import * as path from 'path';
+import { Letter } from './WordleTrie';
+import { Transform } from 'stream';
+
+type LetterCounts = Record<Letter, number>;
+
+const valid_letters = 'abcdefghijklmnopqrstuvwxyz'.split('');
+
+class FilterFor5LetterAlphaWords extends Transform {
+  _transform(chunk, enc, done) {
+    const words = chunk.toString('utf8').split(/\s+/g);
+    const processedChunk = words
```

Comment:
this looks crazy, but what its doing is filtering looking for 5-letter words or 6-letter words that end in `s` ... (primitive since not all pluralizations are just `s`)

we lowercase it and remove the `s` and then validate it is alphabetic

later on in the process we only use words that are in the dictionary, so it fixes any butcher jobs the "stripping the s" does on 6-letter words that have a legitimate s at the end